signing.go

```diff
@@ -151,34 +155,51 @@ func NewMultiSigner(sigs []SigningKey, opts *SignerOptions) (Signer, error) {
 	return signer, nil
 }
 
-// newVerifier creates a verifier based on the key type
-func newVerifier(verificationKey interface{}) (payloadVerifier, error) {
-	switch verificationKey := verificationKey.(type) {
+func NewRSAVerifier(key *rsa.PublicKey) (Verifier, error) {
```

Comment:
Go's idiomatic way is to _"accept interfaces, return structs"_. In this case the `Verifier` interface masks the concrete type, which is unexported. I think it'd be better to return `*RSAEncrypterVerifier` (and similar for other initializers below).

Reply:
Also, this might be relevant here https://github.com/golang/go/wiki/CodeReviewComments#interfaces

Reply:
Hm, that's a very good point, but sort of makes it a larger refactor. I'd want to split those types out into a separate package. And then maybe we don't even need a NewXYZ function for these, and they can just be instantiated via the struct directly? Kinda not sure that's worth it then to refactor this way 🤔 Maybe we should hold onto this idea for v4 and just release v3 now with module support... Bumping versions later should be easy anyway so that's probably fine, if we still want this later we just do v4.